translator/translate/logs/util/filter_retention.go

```diff
@@ -0,0 +1,40 @@
+package util
+
+import (
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/translator"
+	"strings"
+)
+
+func CheckForConflictingRetentionSettings(logConfigs []interface{}, currPath string) []interface{} {
+	configMap := make(map[string]int)
+	for _, logConfig := range logConfigs {
+		if logConfigMap, ok := logConfig.(map[string]interface{}); ok {
+			// skip if retention is not set
+			if retention, ok := logConfigMap["retention_in_days"].(int); ok {
```

Comment:
Because I'm paranoid, where do the `retention_in_days` and `log_group_name` keys come from? If they come from a centralized, public set of constants in another file, I would prefer to reference those. If they're spread out, I still think we should at least pull these out into private scoped constants in this file, like:
```go
const (
    logRetentionKey = "retention_in_days"
    logGroupKey = "log_group_name"
)
```